src/components/codeSnippet/codeSnippet.tsx

```diff
@@ -0,0 +1,84 @@
+import './codeSnippet.css'
+import React, { useEffect, useState } from 'react'
+import { EditorView, minimalSetup } from 'codemirror'
+import { oneDark } from '@codemirror/theme-one-dark'
+import { EditorState } from '@codemirror/state'
+import { Box } from '@mui/system'
+import { Typography } from '@mui/material'
+import { v4 as getUUID } from 'uuid'
+import { LanguageSupport } from '@codemirror/language'
+import ContentPasteIcon from '@mui/icons-material/ContentPaste'
+import { languages } from '@codemirror/language-data'
+import { codeSnippetBgColor } from '../../rusticTheme'
+
+export interface CodeSnippetProps {
+  /** Code that will be displayed */
+  code: string
+  /** Language type needs to be provided so that the right language extension can be used to format code. Language is case-sensitive. Please refer to the [supported languages list](https://codemirror.net/5/mode/). */
+  language: string
+}
+
+export default function CodeSnippet(props: CodeSnippetProps) {
+  const [copied, setCopied] = useState(false)
+  //codeSnippetId is used to identify individual code snippet.
+  const codeSnippetId = getUUID()
+
+  useEffect(() => {
+    const languageObject = languages.find(
+      (language) => language.name === props.language
+    )
+
+    const commonExtensions = [
+      minimalSetup,
+      oneDark,
+      EditorState.readOnly.of(true),
+    ]
+
+    if (languageObject) {
+      languageObject.load().then((module) => {
+        new EditorView({
+          doc: props.code,
+          extensions: [...commonExtensions, module.extension],
+          parent: document.getElementById(codeSnippetId) as HTMLElement,
+        })
+      })
+    } else {
+      new EditorView({
+        doc: props.code,
+        extensions: commonExtensions,
+        parent: document.getElementById(codeSnippetId) as HTMLElement,
+      })
+    }
+  }, [props.language])
```

Comment:
Is this dependency needed?
```suggestion
  }, [])
```